tracks/r/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,57 @@
+### Reasonable solution
+
+This solution is easy to follow, has minimal re-assignments, and uses conditional logic in such a way to make the code more readable.
+
+```r
+bob <- function(input) {
+  trimmed_input <- trimws(input)
+
+  is_shouted_statement <- is_shouting(trimmed_input)
+  is_question_statement <- is_question(trimmed_input)
+
+  if (nchar(trimmed_input) == 0) {
+    "Fine. Be that way!"
+  } else if (is_shouted_statement && is_question_statement) {
+    "Calm down, I know what I'm doing!"
+  } else if (is_question_statement) {
+    "Sure."
+  } else if (is_shouted_statement) {
+    "Whoa, chill out!"
+  } else {
+    "Whatever."
+  }
+}
+
+is_question <- function(statement) { grepl("\\?$", statement) }
+is_shouting <- function(statement) { grepl("^[A-Z]+$", letters_only(statement)) }
+letters_only <- function(string) { gsub("[^a-zA-Z]", "", string) }
+```
+
+The learning goal here is to get comfortable with regular-expressions, pattern matching, and common string operations. Secondarily, having students check multiple conditions in an orderly fashion bolsters their conditional flow control.
+
+
+### Common suggestions
+
+- Readability can easily be degraded with dense conditional statements, or especially when trying to index on a vector using a Boolean vector (although decent examples exist, like the one below); have students try to make their answer as intelligible as possible.
```

Comment:
I would _try_ to write this similarly as `predicate -> action`, just like the second item in this list:
```markdown
- If a student uses a Boolean vector, ...
- If a student uses dense conditional statements, ....
```

The content is great, but we probably want to aid the mentor finding the suggestions as quickly as possible.